### ISSUE TITLE
plot your connectome as interactive visualization.

### ISSUE THREAD

**Issue by russelljjarvis** [russelljjarvis/teili#1]
@pabloabur 

I was thinking of trying to visualize your predictive coding network structure network as an interactive graph. I think I would need to wrangle it into various `networkx` `nx` directed graphs. Possibly each seperate layer is its own graph and I would need to stitch them all together.

I think these are the teili methods I would use:
https://github.com/russelljjarvis/teili/commit/0a14380a85fb1345f27ed29cfde61f95d96aac80

**Comment by pabloabur** [comment]
Yes, I think that would be nice. Maybe we could use the existing _plot()_ method to plot intralaminar structure (which is simpler), and use these graphs to plot interlaminar and columnar connections.

**Comment by russelljjarvis** [comment]
Sure, can I apply the plot method to get an adjacency matrix for all layers? Or does that method only deal with one intralaminar structure for respective layers?

**Comment by pabloabur** [comment]
You can plot any Connection group, but plotting for all layers would be a mess because you show all connections between all neurons. So in a hierarchical large-scale network, you would only see a bunch of lines. If pre and post populations are small, it is ok. You can see a few examples [here](https://brian2.readthedocs.io/en/stable/resources/tutorials/2-intro-to-brian-synapses.html#more-complex-connectivity)

**Comment by russelljjarvis** [comment]
I looked at the layer to layer connectivity plots. I think sometimes we might want that. 

How much does your predictive coding network model deviate from the Potjan's wiring plan.

I wonder if interactive parallel coordinate plots might be okay.
If you look at this [plot](https://bbp.epfl.ch/nmc-portal/microcircuit.html) you can toggle scale. So you can look at synapse level connections, or population level connections.

The layer to layer plots are also known as parallel coordinate plots you can see some more [here](https://www.google.com/search?q=parallel+coordinates+plot&sxsrf=ALiCzsbALRFukDIDQ7v5PgpiCTb2bqZ7hA:1651193032304&source=lnms&tbm=isch&sa=X&sqi=2&ved=2ahUKEwj0pZWxhbj3AhWNyYsKHUVxC4QQ_AUoAXoECAIQAw&biw=1592&bih=652&dpr=1)

**Comment by russelljjarvis** [comment]
I wonder if interactive parallel coordinate plots might be okay.

**Comment by pabloabur** [comment]
Yes, that would be helpful. Especially if you can select which connection you want to see

**Comment by russelljjarvis** [comment]
Hive plots are a type of rotated linear axis parallel coordinate plot.

There is an an example of a hive plot here:

https://share.streamlit.io/russelljjarvis/odor2action/app.py
